plugins/woocommerce/src/Internal/Traits/OrderAttributionMeta.php

```diff
@@ -375,4 +396,52 @@ private function get_field_description( string $field ): string {
 		 */
 		return (string) apply_filters( 'wc_order_attribution_field_description', $description, $field );
 	}
+
+	/**
+	 * Get the order history for the customer (data matches Customers report).
+	 *
+	 * @param mixed $customer_identifier The customer ID or billing email.
+	 *
+	 * @return array Order count, total spend, and average spend per order.
+	 */
+	private function get_customer_history( $customer_identifier ): array {
+
+		// Get the valid customer orders.
+		$args = array(
+			'limit'  => - 1,
+			'return' => 'objects',
+			// Don't count cancelled, on-hold, or failed orders.
+			'status' => array( 'pending', 'processing', 'on-hold', 'completed', 'refunded' ),
+			'type'   => 'shop_order',
+		);
+
+		// If the customer_identifier is a valid ID, use it. Otherwise, use the billing email.
+		if ( is_numeric( $customer_identifier ) && $customer_identifier > 0 ) {
+			$args['customer_id'] = $customer_identifier;
+		} else {
+			$args['billing_email'] = $customer_identifier;
+			$args['customer_id']   = 0;
+		}
+
+		$orders = wc_get_orders( $args );
+
+		// Populate the order_count and total_spent variables with the valid orders.
+		$order_count = 0;
+		$total_spent = 0;
+		foreach ( $orders as $order ) {
+			$order_count ++;
+			$total_spent += $order->get_total();
+
+			// Remove any refunded amount from the total_spent.
+			foreach ( $order->get_refunds() as $refund ) {
+				$total_spent += $refund->get_total();
+			}
+		}
```

Comment:
Should we maybe use `get_total_refunded` Order class method instead of calculating manually?

I think that in case the order is fully refunded it should not be counted towards `$order_count`

```suggestion
			$order_total = $order->get_total() - $order->get_total_refunded();
			
			if ( $order_total ) {
				$order_count++;
				$total_spent += $order_total;
			}
		}
```

Reply:
Hmm, there may be value to bumping the order count even in case the order if fully refunded. We can leave it as it is  in case you want to keep it.

Reply:
> Should we maybe use `get_total_refunded` Order class method instead of calculating manually?

Nice catch.

>  order is fully refunded it should not be counted towards $order_count

I believe that I tested a few different configurations and tried to make this exactly match the numbers for the customer on `wp-admin/admin.php?page=wc-admin&path=/customers`. I'll do another check to make sure they line up before merging.

Reply:
Just a heads up that, in the end, refunded orders to count towards the order_count, and the `get_total_refunded` method is accurate. 7369d49c432905424bb7fe918f2f12316af6513b